### PR TITLE
Added rel="noopener" to target="_blank" hrefs

### LINKED ISSUE #315: [security] target="_blank" without rel="noopener"
https://github.com/go-gitea/gitea/blob/bad1bc6/models/repo.go#L502

If you have links to another origin, you should use `rel="noopener"`, especially if they open in a new tab/window.

```html
<a href="http://example.com" target="_blank" rel="noopener">
   Example site
</a>
```

Without this, the new page can access your window object via window.opener. Thankfully the origin security model of the web prevents it reading your page, but no-thankfully some legacy APIs mean it can navigate your page to a different URL using `window.opener.location = newURL`.

Refs
---
* https://mathiasbynens.github.io/rel-noopener/
* https://jakearchibald.com/2016/performance-benefits-of-rel-noopener/
* https://sites.google.com/site/bughunteruniversity/nonvuln/phishing-with-window-opener
* https://github.com/danielstjules/blankshield
* https://www.jitbit.com/alexblog/256-targetblank---the-most-underestimated-vulnerability-ever/
* https://dev.to/ben/the-targetblank-vulnerability-by-example
* https://github.com/cure53/H5SC#html5-security-cheatsheet
* https://github.com/cure53/DOMPurify

### PR DESCRIPTION
**Pull request by Bwko**
Partially fixes #315
I've just ran a replace on gitea, replacing `target="_blank"` with `target="_blank" rel="noopener"`

### CODE REVIEW

**Comment by lunny** [comment]
But maybe only external URL needs this property?

**Comment by Bwko** [comment]
Yes but it can't hurt to add to internal URLs , since just replacing all the _blanks was the easiest solution

File: templates/base/footer.tmpl [hunk 1]
```diff
@@ -11,9 +11,9 @@
 			</div>
 			<div class="ui right links">
 				{{if .ShowFooterBranding}}
-					<a target="_blank" href="https://github.com/go-gitea/gitea"><i class="fa fa-github-square"></i><span class="sr-only">GitHub</span></a>
-					<a target="_blank" href="https://twitter.com/gogitservice"><i class="fa fa-twitter"></i><span class="sr-only">Twitter</span></a>
-					<a target="_blank" href="http://weibo.com/gogschina"><i class="fa fa-weibo"></i><span class="sr-only">Sina Weibo</span></a>
+					<a target="_blank" rel="noopener" href="https://github.com/go-gitea/gitea"><i class="fa fa-github-square"></i><span class="sr-only">GitHub</span></a>
+					<a target="_blank" rel="noopener" href="https://twitter.com/gogitservice"><i class="fa fa-twitter"></i><span class="sr-only">Twitter</span></a>
```
Comment (tboerger) [inline templates/base/footer.tmpl]: We need to drop that

File: templates/base/footer.tmpl [hunk 2]
```diff
@@ -11,9 +11,9 @@
...
+					<a target="_blank" rel="noopener" href="http://weibo.com/gogschina"><i class="fa fa-weibo"></i><span class="sr-only">Sina Weibo</span></a>
```
Comment (tboerger) [inline templates/base/footer.tmpl]: This as well

File: templates/base/head.tmpl [hunk 3]
```diff
@@ -113,7 +113,7 @@
 													<i class="octicon octicon-settings"></i>
 													{{.i18n.Tr "your_settings"}}<!-- Your settings -->
 												</a>
-												<a class="item" target="_blank" href="https://gogs.io/docs" rel="noreferrer">
+												<a class="item" target="_blank" rel="noopener" href="https://gogs.io/docs" rel="noreferrer">
```
Comment (tboerger) [inline templates/base/head.tmpl]: https://docs.gitea.io

File: templates/base/head.tmpl [hunk 4]
```diff
@@ -137,7 +137,7 @@
 
 								{{else}}
 
-									<a class="item" target="_blank" href="https://gogs.io/docs" rel="noreferrer">{{.i18n.Tr "help"}}</a>
+									<a class="item" target="_blank" rel="noopener" href="https://gogs.io/docs" rel="noreferrer">{{.i18n.Tr "help"}}</a>
```
Comment (tboerger) [inline templates/base/head.tmpl]: Same as above

File: templates/home.tmpl [hunk 5]
```diff
@@ -20,15 +20,15 @@
 					<i class="octicon octicon-flame"></i> Einfach zu installieren
 				</h1>
 				<p class="large">
-					Starte einfach <a target="_blank" href="https://gogs.io/docs/installation/install_from_binary.html">die Anwendung</a> für deine Plattform. Gitea gibt es auch für <a target="_blank" href="https://github.com/go-gitea/gitea/tree/master/docker">Docker</a>, <a target="_blank" href="https://github.com/geerlingguy/ansible-vagrant-examples/tree/master/gogs">Vagrant</a> oder als <a target="_blank" href="https://gogs.io/docs/installation/install_from_packages.html">Installationspaket</a>.
+					Starte einfach <a target="_blank" rel="noopener" href="https://gogs.io/docs/installation/install_from_binary.html">die Anwendung</a> für deine Plattform. Gitea gibt es auch für <a target="_blank" rel="noopener" href="https://github.com/go-gitea/gitea/tree/master/docker">Docker</a>, <a target="_blank" rel="noopener" href="https://github.com/geerlingguy/ansible-vagrant-examples/tree/master/gogs">Vagrant</a> oder als <a target="_blank" rel="noopener" href="https://gogs.io/docs/installation/install_from_packages.html">Installationspaket</a>.
```
Comment (tboerger) [inline templates/home.tmpl]: Same as above

File: templates/repo/issue/labels.tmpl [hunk 6]
```diff
@@ -43,7 +43,7 @@
 						<div class="ui attached left aligned segment">
 							<h4 class="ui header">
 								{{.i18n.Tr "repo.issues.label_templates.title"}}
-								<a target="_blank"
+								<a target="_blank" rel="noopener"
 								   href="https://discuss.gogs.io/t/how-to-use-predefined-label-templates/599">
```
Comment (tboerger) [inline templates/repo/issue/labels.tmpl:47]: This link also won't work for us

**Review by tboerger — COMMENTED** [review]
Not entirely related to the topic but we need to change that.

**Comment by thibaultmeyer** [comment]
LGTM

**Comment by tboerger** [comment]
LGTM

**Pull request merged**